src/commands/Utility/spotify.js

```diff
@@ -1,95 +1,136 @@
-            }, this);
-
-            let duration_m = Math.floor(data.duration_ms / 60000);
-            let duration_s = ((data.duration_ms % 60000) / 1000).toFixed(0);
-
-            let embed = bot.utils.embed(
-                'Listen now!',
-                `${msg.author.username} is currently listening to`,
-                [
-                    {
-                        name: 'Artist(s)',
-                        value: `${artists.join(', ')}`,
-                    },
-                    {
-                        name: 'Title',
-                        value: `${data.name}`,
-                    },
-                    {
-                        name: 'Explict',
-                        value: `${(data.explict) ? 'yes' : 'no'}`,
-                    },
-                    {
-                        name: 'Popularity',
-                        value: `${data.popularity} %`,
-                    },
-                    {
-                        name: 'Duration',
-                        value: (duration_s == 60 ? (duration_m + 1) + ':00' : duration_m + ':' + (duration_s < 10 ? '0' : '') + duration_s) + 'min',
-                    },
-                    {
-                        name: 'Markets',
-                        value: `Song is available on ${data.available_markets.length} markets`,
-                    },
-                ],
-                {
-                    inline: true,
-                    footer: 'Spotify Track Info',
-                    color: [30, 215, 96]
-                });
-
-            embed.setThumbnail(`${data['album']['images'][2]['url']}`);
-            embed.setURL(`${data['external_urls']['spotify']}`);
-            msg.delete();
-            msg.channel.send({ embed });
-
-        } else {
-            msg.error(`No info found for ${input}`);
-        }
+    }, function(err) {
+        throw `Something went wrong! ${err}`;
+    });
+}
+
+function getArtistEmbed(msg, spotifyId, withPreview) {
+    spotifyApi.getArtist(spotifyId)
+    .then(function(data) {
+        let apiResponse = data.body;
+        
+        const embed = new RichEmbed()
+        .setColor([30,215,96])
+        .setThumbnail(apiResponse.images[0].url)
+        .setAuthor('Click to listen on Spotify','https://image.flaticon.com/icons/png/512/174/174872.png', apiResponse.external_urls.spotify)
+        .addField('Artist Name', apiResponse.name, true)
+        .addField('Followers',  apiResponse.followers.total.toLocaleString(), true)
+        .addField('Popularity',  apiResponse.popularity + '%', true)
+        .addField('Genres',  ((apiResponse.genres.length > 0)?apiResponse.genres.join(', '):'unknown'), true);
+
+        msg.edit({content:'',embed:embed});
+        
+        if(withPreview) msg.channel.send(`Got spotify? Click below to play! ${apiResponse.external_urls.spotify}`);
 
-    }).catch(err => {
-        msg.error('SpotifyAPI returned an error. See console.');
-        bot.logger.severe(err);
+    }, function(err) {
```

Comment:
Please use lambda notation (`(...) => { ... }`) instead of function notation (`function (...) { ... }`) for the sake of consistency in code style.

Also, again, I would prefer the `.catch` syntax for promises rather than passing two callbacks to `.then`.